common/djangoapps/student/views.py

```diff
@@ -678,15 +679,22 @@ def dashboard(request):
         'ACTIVATION_EMAIL_SUPPORT_LINK', settings.ACTIVATION_EMAIL_SUPPORT_LINK
     ) or settings.SUPPORT_SITE_LINK
 
-    # get the org whitelist or the org blacklist for the current site
+    # Get the org whitelist or the org blacklist for the current site
     site_org_whitelist, site_org_blacklist = get_org_black_and_whitelist_for_site(user)
     course_enrollments = list(get_course_enrollments(user, site_org_whitelist, site_org_blacklist))
 
+    # Get the entitlements for the user and a mapping to all available sessions for that entitlement
+    course_entitlements = list(CourseEntitlement.objects.filter(user=user))
+    course_entitlement_available_sessions = {
+        str(entitlement.uuid): get_course_runs_for_course(str(entitlement.course_uuid))
+        for entitlement in course_entitlements
+    }
```

Comment:
Do we need to check for expired entitlements here?

Reply:
We were going to weave that in to the follow up story with Jeff for all the expired behavior.